src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-up/index.mdx

```diff
@@ -527,3 +527,363 @@ export default function App() {
+
+```swift
+// Sign up using an email
+func signUp(username: String, email: String) async {
+    let userAttributes = [
+        AuthUserAttribute(.email, value: email)
+    ]
+    let options = AuthSignUpRequest.Options(userAttributes: userAttributes)
+    do {
+        let signUpResult = try await Amplify.Auth.signUp(
+            username: username,
+            options: options
+        )
+        if case let .confirmUser(deliveryDetails, _, userId) = signUpResult.nextStep {
+            print("Delivery details \(String(describing: deliveryDetails)) for userId: \(String(describing: userId)))")
+        } else {
+            print("SignUp Complete")
+        }
+    } catch let error as AuthError {
+        print("An error occurred while registering a user \(error)")
+    } catch {
+        print("Unexpected error: \(error)")
+    }
+}
+
+// Confirm sign up with the OTP received
+func confirmSignUp(for username: String, with confirmationCode: String) async {
+    do {
+        let confirmSignUpResult = try await Amplify.Auth.confirmSignUp(
+            for: username,
+            confirmationCode: confirmationCode
+        )
+        print("Confirm sign up result completed: \(confirmSignUpResult.isSignUpComplete)")
+    } catch let error as AuthError {
+        print("An error occurred while confirming sign up \(error)")
+    } catch {
+        print("Unexpected error: \(error)")
+    }
+}
+```
+
+</Block>
+
+<Block name="Combine">
+
+```swift
+// Sign up using an email
+func signUp(username: String, email: String) -> AnyCancellable {
+    let userAttributes = [
+        AuthUserAttribute(.email, value: email)
+    ]
+    let options = AuthSignUpRequest.Options(userAttributes: userAttributes)
+    let sink = Amplify.Publisher.create {
+        try await Amplify.Auth.signUp(
+            username: username,
+            options: options
+        )
+    }.sink {
+        if case let .failure(authError) = $0 {
+            print("An error occurred while registering a user \(authError)")
+        }
+    }
+    receiveValue: { signUpResult in
+        if case let .confirmUser(deliveryDetails, _, userId) = signUpResult.nextStep {
+            print("Delivery details \(String(describing: deliveryDetails)) for userId: \(String(describing: userId)))")
+        } else {
+            print("SignUp Complete")
+        }
+    }
+    return sink
+}
+
+// Confirm sign up with the OTP received
+func confirmSignUp(for username: String, with confirmationCode: String) -> AnyCancellable {
+    Amplify.Publisher.create {
+        try await Amplify.Auth.confirmSignUp(
+            for: username,
+            confirmationCode: confirmationCode
+        )
+    }.sink {
+        if case let .failure(authError) = $0 {
+            print("An error occurred while confirming sign up \(authError)")
+        }
+    }
+    receiveValue: { _ in
+        print("Confirm signUp succeeded")
+    }
+}
+```
+
+</Block>
+
+</BlockSwitcher>
+
+</InlineFilter>
+
+### WebAuthn Passkeys
```

Comment:
Should be able to remove this information as it's not possible to sign-up via passkey - passkeys can only be registered after creating an account and signing in with one of the other methods.